addons/account_edi/models/account_journal.py

```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    edi_ids = fields.Many2many(comodel_name='account.edi', string='EDI Format', help='When sending an invoice, the following XML will be attached or embedd to the PDF.')
+
+    def _create_invoice_from_attachment(self, attachment):
```

Comment:
This method shouldn't be defined on `account.journal`, it has nothing to do with it

Reply:
The import from the mail alias is implemented in `account.journal`. This is not a new stuff, just an additional hook

Reply:
It's when you use the 'upload' button on a journal.